src/components/Event/utils/dateTimeHelpers.ts

```diff
@@ -83,3 +88,35 @@ export const dtDate = (d: Dayjs) => d.format("YYYY-MM-DD");
 
 /** Extract time “HH:mm” */
 export const dtTime = (d: Dayjs) => d.format("HH:mm");
+
+/**
+ * Parse flexible time string input into Dayjs object
+ * Handles formats like: "HH:mm", "HHmm", "Hmm", "HH", "H"
+ * Also handles 3-digit input normalization (e.g. "830" -> "08:30")
+ */
+export function parseTimeInput(
+  value: string,
+  currentDate: Dayjs | null
+): Dayjs | null {
+  let trimmed = value.trim();
+  if (!trimmed) {
+    return null;
+  }
+
+  // Normalize 3-digit input (e.g. "830" → "0830" → 08:30, "123" → "0123" → 01:23)
+  if (/^\d{3}$/.test(trimmed)) {
+    trimmed = trimmed.padStart(4, "0");
+  }
+
+  const parsed = dayjs(trimmed, TIME_PARSE_FORMATS, true);
```

Comment:
What does `true` means?
It's a magic constant and generally giving it a name eg with a variable or a constant makes it much easier for the non specialist to know what is going on.